pds_doi_service/core/outputs/osti/osti_record.py

```diff
@@ -0,0 +1,145 @@
+#
+#  Copyright 2021, by the California Institute of Technology.  ALL RIGHTS
+#  RESERVED. United States Government Sponsorship acknowledged. Any commercial
+#  use must be negotiated with the Office of Technology Transfer at the
+#  California Institute of Technology.
+#
+
+"""
+==============
+osti_record.py
+==============
+
+Contains classes used to create OSTI-compatible labels from Doi objects in memory.
+"""
+
+import html
+import json
+from datetime import datetime
+from os.path import exists
+
+import pystache
+from pkg_resources import resource_filename
+
+from pds_doi_service.core.entities.doi import Doi, ProductType
+from pds_doi_service.core.outputs.doi_record import (DOIRecord,
+                                                     CONTENT_TYPE_XML,
+                                                     CONTENT_TYPE_JSON,
+                                                     VALID_CONTENT_TYPES)
+from pds_doi_service.core.util.general_util import get_logger
+
+logger = get_logger(__name__)
+
+
+class DOIOstiRecord(DOIRecord):
+    """
+    Class used to create a DOI record suitable for submission to the OSTI
+    DOI service.
+
+    This class supports output of DOI records in both XML and JSON format.
+    """
+    def __init__(self):
+        """Creates a new DOIOstiRecord instance"""
+        # Need to find the mustache DOI templates
+        self._xml_template_path = resource_filename(
+            __name__, 'DOI_IAD2_template_20200205-mustache.xml'
+        )
+        self._json_template_path = resource_filename(
+            __name__, 'DOI_IAD2_template_20210216-mustache.json'
+        )
```

Comment:
I get giddy when I see people use `pkg_resources` 😊 this is excellent 🥳

Reply:
🤜 🤛